### PR TITLE
Changes default behaviour to ignore archived and deleted repos

### PR DESCRIPTION
**Pull request by hbontempo-br**
## Description

Changes default behavior of ReposModel to deleted and archived repos.
This repos are going to remain on database but should be ignored everywhere in the application

## Related Issue

Related to #1704
Should deal with #1735 / #1545

## Motivation and Context

Should prevent "bad" repos to be presented to users

## How Has This Been Tested?

Automated tests + rubocop analysis (locally)

## Screenshots (if appropriate):
--

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: app/models/repo.rb [hunk 1]
```diff
@@ -24,6 +24,9 @@ class Repo < ActiveRecord::Base
 
   CLASS_FOR_DOC_LANGUAGE = { "ruby" => DocsDoctor::Parsers::Ruby::Yard }
 
+  default_scope -> { where(removed_from_github: false) }
+  default_scope -> { where(archived: false) }
```
Comment (schneems) [inline app/models/repo.rb:28]: I'm strongly opposed to `default_scope` but enumerating why is difficult. It's just one of those things that I've built up scar tissue from over the years. It saves a few minutes now and really hurts later when debugging etc. 

I'm asking on twitter for alternatives. So far I've got:

- Make it a regular scope and call it everywhere. (not so different than `with_some_issues` which is used heavily https://github.com/codetriage/CodeTriage/blob/aa03ffcd56f57c2b12ffb48a752ae27ec3ea93b7/app/models/repo.rb#L176
- Split up logic into different classes. So you might have a AllRepo  class and an ArchivedRepo class etc.  I'm not sure what we would call the class that ignores deleted and archived, maybe `ActiveRepo` class ? 
- Might get more replies on twitter

I think I would lean towards making it a regular scope and applying it as needed.

**Review by schneems — CHANGES_REQUESTED** [review]
Please change out `default_scope` for something else.

File: app/models/repo.rb (hunk 1 above)
Comment (hbontempo-br) [reply app/models/repo.rb:28]: Thanks for the feedback!
I don't have much experience, will take a look in the tweeter responses and search a little bit.
I will come back soon with a alternative.

Comment (hbontempo-br) [reply app/models/repo.rb]: I read a bit and `default_scope` is really broadly hated in the community.
Changed to a 'normal' scope, it's the simplest alternative.

**Comment by hbontempo-br** [comment]
PR ended up being bigger than originally expected.
Could break up into smaller ones if you prefer.

**Review by schneems — APPROVED** [review]

**Comment by schneems** [comment]
Thanks for working on this. Sorry for the EXTREME delay. Looks great. I corrected a merge conflict. re-running tests now.

**Pull request merged**